src/core/logical_layers/renderers/stylesConfigs/mcda/mcdaStyle.ts

```diff
@@ -85,7 +85,7 @@ function sentimentPaint({
     'fill-color': [
       'let',
       'mcdaResult',
-      ['to-number', mcdaResult, -9999], // falsy values become -9999
```

Comment:
Old code was converting any faulty value (like NaN or null) into -9999 which caused them to be colored instead of being transparent